### ISSUE TITLE
Error is empty array when requested range is not satisfiable

### ISSUE THREAD

**Issue by Curetix** [supabase/postgrest-js#324]
# Bug report

## Describe the bug

When using the `range()` filter on a select query, it is possible that the range will not be satisfiable. In that case the fetch library throws an error because of status code 416. This error is logged to the console, but not returned in the `error` object of the response, which is just an empty array.

Also, the TypeScript types do not reflect that this is a possibility.

## To Reproduce

This is rougly my code:

```typescript
try {
  const { data, error, count } = await supabase
    .from('games')
    .select('*', { count: 'exact' })
    .order(filter.sortBy ? filter.sortBy : 'updated_at', { ascending: filter.sortOrder === 'ascending' })
    .range(filter.pageSize * (filter.page - 1), filter.pageSize * filter.page - 1);

  if (error) {
    console.error('[Games] Error during fetch', error);
  } else {
    this.games = data || [];
    this.totalGamesCount = count || 0;
  }
} catch (error) {
  console.error('[Games] Error during fetch', error);
}
```

## Expected behavior

If the range is out-of-bounds, that should be properly reported in the `error` object.

## Screenshots

The console output in such a situation:

![image](https://user-images.githubusercontent.com/15160542/189072943-3d4c80cd-38bb-4059-9a8d-59b7cc8bb3a2.png)

## System information

- Browser: Brave
- Version of supabase-js: 2.0.0-rc.9
- Version of Node.js: v16.16.0

**Comment by soedirgo** [comment]
Thanks for reporting this, seems like a similar issue to #295. I'll close this in favor of that one.

**Comment by steve-chavez** [comment]
This is different from #295 because there is a error message on the response body coming from the server.

I think we need to correct something on `postgrest-js` side.

**Comment by soedirgo** [comment]
There's no error message in this case though:
```
❯ xhs get [...].supabase.co/rest/v1/t 'Authorization:Bearer [...]' 'apikey:[...]' 'Accept:application/json' 'Prefer:count=exact' offset==1000
HTTP/2.0 416 Range Not Satisfiable
access-control-allow-origin: *
alt-svc: h3=":443"; ma=86400, h3-29=":443"; ma=86400
cf-cache-status: DYNAMIC
cf-ray: 747e31fdab2a18ac-SIN
content-location: /t?offset=1000
content-profile: public
content-range: */5
content-type: application/json; charset=utf-8
date: Fri, 09 Sep 2022 07:26:22 GMT
sb-gateway-version: 1
server: cloudflare
vary: Accept-Encoding
via: kong/2.8.1
x-kong-proxy-latency: 1
x-kong-upstream-latency: 4

[]
```

**Comment by soedirgo** [comment]
@steve-chavez should we keep this one open? Or should we track it on the other issue and wait for upstream fix?

**Comment by steve-chavez** [comment]
This one is also an upstream fix(different from the other issue), it's close to being merged https://github.com/PostgREST/postgrest/pull/2472

**Comment by Curetix** [comment]
> This one is also an upstream fix(different from the other issue), it's close to being merged [PostgREST/postgrest#2472](https://github.com/PostgREST/postgrest/pull/2472)
> 
> **Edit**: Merged 🚀

Thanks for the fix!
